### PR TITLE
Implement the melkman anlgorithm for computing convex hulls

### PR DESCRIPTION
**Pull request by SimonLammer**
### **Describe your change:**

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

~~Since the file didn't contain type hints at all, I didn't add them for this algorithm.~~
Is this file the correct place for the algorithm? It is not a divide and conquer algorithm, but it is a convex_hull algorithm.

### CODE REVIEW

File: divide_and_conquer/convex_hull.py [hunk 1]
```diff
@@ -411,6 +411,77 @@ def _construct_hull(points, left, right, convex_set):
             _construct_hull(candidate_points, extreme_point, right, convex_set)
 
 
+def convex_hull_melkman(points):
```
Comment (poyea) [inline divide_and_conquer/convex_hull.py:414]: Maybe specify the type for points? You may see our typing guidelines.

Comment (poyea) [reply divide_and_conquer/convex_hull.py:414]: You may also do this for other functions in this file.

Comment (SimonLammer) [reply divide_and_conquer/convex_hull.py]: done

**Comment by SimonLammer** [comment]
@poyea Is there anything else I should to revise?

**Comment by SimonLammer** [comment]
@poyea ?

**Review by 53jk1 — APPROVED** [review]

**Review by poyea — APPROVED** [review]
Thank you for your pull request!🤩

**Comment by poyea** [comment]
@SimonLammer Add a `hacktoberfest` tag in your fork to make this valid for Hacktoberfest. This should work.

**Pull request merged**